src/main/java/at/hannibal2/skyhanni/features/garden/pests/PesthunterProfit.kt

```diff
@@ -0,0 +1,200 @@
+        val pestsCost = getPestsCost(item)
+        val profitPerPest = if (pestsCost > 0) profit / pestsCost else 0.0
+        val color = if (profitPerPest > 0) "§6" else "§c"
+
+        val hover = listOf(
+            itemName.replace("[Lvl {LVL}]", "[Lvl 1]"),
+            "",
+            "§7Item price: §6${itemPrice.shortFormat()} ",
+            "§7Material cost: §6${totalCost.shortFormat()} ",
+            "§7Final profit: §6${profit.shortFormat()} ",
+            "§7Profit per pest: §6${profitPerPest.shortFormat()} ",
+        )
+
+        table.add(
+            DisplayTableEntry(
+                itemName.replace("[Lvl {LVL}]", "[Lvl 1]"), // show level 1 hedgehog instead of level 100
+                "$color${profitPerPest.shortFormat()}",
+                profitPerPest,
+                internalName,
+                hover,
+                highlightsOnHoverSlots = listOf(slot),
+            ),
+        )
+
+        if (bestPesthunterTrade.isEmpty() || profitPerPest > (
+                bestPesthunterTrade.maxByOrNull {
+                    it.coinsPerPest
+                }?.coinsPerPest ?: 0.0
+                )
+        ) {
+            bestPesthunterTrade.clear()
+            bestPesthunterTrade.add(PesthunterTrade(internalName, profitPerPest))
+        }
+    }
+
+    private fun getRequiredItems(item: ItemStack): MutableList<String> {
+        val items = mutableListOf<String>()
+        var next = false
+
+        for (line in item.getLore()) {
+            when {
+                line == "§7Cost" -> {
+                    next = true
+                }
+                next -> {
+                    if (line.isBlank()) {
+                        next = false
+                    } else {
+                        pestCostPattern.matchMatcher(line.replace(",", "")) { break }
+                        items.add(line.replace("§8 ", " §8"))
+                    }
+                }
+            }
+        }
+
+        return items
+    }
+
+    private fun getFullCost(requiredItems: List<String>): Double {
+        var otherItemsPrice = 0.0
+        for (itemName in requiredItems) {
+            val pair = ItemUtils.readItemAmount(itemName)
+            if (pair == null) {
+                ErrorManager.logErrorStateWithData(
+                    "Error in Pesthunter Profit", "Could not read item amount",
+                    "itemName" to itemName,
+                )
+                continue
+            }
+
+            val (name, amount) = pair
+            otherItemsPrice += NEUInternalName.fromItemName(name).getPrice() * amount
+        }
+        return otherItemsPrice
+    }
+
+    private fun getPestsCost(item: ItemStack): Int {
+        val lore = item.getLore()
+        for (line in lore) {
+            val pestCost = line.replace(",", "")
+            pestCostPattern.matchMatcher(pestCost) {
+                return group("pests")?.toInt() ?: 0
+            }
+        }
+        return 0
```

Comment:
we have utils methods for finding the first match in lore and for converting numbers with commas in them to numbers. See RegexUtils and NumberUtil.
This also means the repo pattern should be updated to accept numbers with commas